### DISCUSSION TITLE
database disk image is malformed

### DISCUSSION THREAD

**Discussion by roelvdh** [SBFspot/SBFspot#500] [Q&A]
Hi, 
I needed to load some old data from my SMA inverter using
`pi@SolarPi:/usr/local/bin/sbfspot.3 $ /usr/local/bin/sbfspot.3/SBFspot -ad45 -am2 -ae2 -v -finq -nocsv`
That seemed OK, but in the end I found a disturbing error:
```
********************
* ArchiveDayData() *
********************
startTime = 617C6EE0 -> 30/10/2021 00:00:00
**********************
* ArchiveMonthData() *
**********************
startTime = 61A755B0 -> 01/12/2021 12:00:00
**********************
* ArchiveMonthData() *
**********************
startTime = 617FC8B0 -> 01/11/2021 12:00:00
Reading events: 2021-Dec-01
Reading events: 2021-Nov-01
[2021-12-13 11:41:30.068] Error: [event_data]sqlite3_step() returned: 'database disk image is malformed'
[2021-12-13 11:41:30.074] Error: [event_data]Transaction failed. Rolling back now...: 'database disk image is malformed'
Mon Dec 13 11:41:30 2021: INFO: Done.
pi@SolarPi:/usr/local/bin/sbfspot.3 $
```

Any idea how to solve this?
Thanks, Roel

**Comment by sbf-** [comment]
Running an integrity check should return OK
```
sqlite3 ~/smadata/SBFspot.db
SQLite version 3.27.2 2019-02-25 16:06:06
Enter ".help" for usage hints.
sqlite> PRAGMA integrity_check;
ok
sqlite> 
```
Most of the time the db is corrupt and if you don't have a backup you will have to start with a new, empty one.
You can try the '.dump' command as well

**Comment by roelvdh** [comment]
Thank you! Result:

sqlite> PRAGMA integrity_check;
*** in database main ***
Page 3707: btreeInitPage() returns error code 11
Page 5259 is never used
Page 5260 is never used
Page 5261 is never used
Page 5262 is never used
Page 5263 is never used
Error: database disk image is malformed
sqlite>

Any chance of recovery? Possible reason is: I had to temporarily remove
from the cfg file 1 of my 2 inverters, as it is currently not operative and
reloading data failed because of the missing communication link. I had no
problems before that action. The second inverter will be reinstated after
repair.

Op ma 13 dec. 2021 om 16:08 schreef SBF ***@***.***>:
